transformer_engine/pytorch/utils.py

```diff
@@ -305,3 +305,18 @@ def devices_match(device1: torch.device, device2: torch.device) -> bool:
             index2 = torch.cuda.current_device()
         return index1 == index2
     return device1 == device2
+
+
+TORCH_MAJOR = int(torch.__version__.split(".")[0])
+TORCH_MINOR = int(torch.__version__.split(".")[1])
+if TORCH_MAJOR == 2 and TORCH_MINOR >= 4:
```

Comment:
I think it would be cleaner to define a single function and do the torch version check inside it to call either `get_autocast_dtype` or `get_autocast_gpu_dtype` as opposed to doing the version check separately and defining 2 separate functions like this.

Reply:
Updated.